sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -111,6 +115,41 @@ object RapidsPluginUtils extends Logging {
     }
   }
 
+  private def detectMultipleJar(propName: String, jarName: String, propsName: Seq[String], 
+      conf: RapidsConf): Unit = {
+    val classloader = ShimLoader.getShimClassLoader()
+    val possibleRapidsJarURLs = classloader.getResources(propName).asScala.toSet
+    val (rapidsJarsURLs, rapidsJarsVers) = possibleRapidsJarURLs.flatMap { 
+      url => {
+        val ver = scala.io.Source.fromInputStream(url.openStream()).mkString("") 
+        if (propsName.exists(ver.contains(_))) {
+          Some((url, ver))
+        } else {
+          None
+        }
+      }
+    }.unzip
+    lazy val rapidsJarsMsg = rapidsJarsURLs.toList.map(_.toString.split("!").head).mkString(",")
+    lazy val rapidsJarsVersMsg = rapidsJarsVers.toList.mkString(",")
+    lazy val msg = s"Multiple $jarName jars found in the classpath: $rapidsJarsMsg, please make " +
+        s"sure there is only one $jarName jar in the classpath. If it is impossible to fix the " +
+        s"classpath you can suppress the error by setting ${RapidsConf.ALLOW_MULTIPLE_JARS.key}" + 
+        s"to true. Version info: \n$rapidsJarsVersMsg"
+    if (conf.allowMultipleJars) {
+      if (rapidsJarsURLs.size > 1) {
+        logWarning(msg)
+      }
+    } else {
+      require(rapidsJarsURLs.size <= 1, msg)
```

Comment:
When can we get a `rapidsJarsURLs` < 1 ? I am not sure whether this case is allowed.

Reply:
It is not going to happen, I made it `==1` now so it will fail if it does happen somehow.